File: app/src/main/java/com/photo/editor/picskills/photoeditorpro/activities/StatusActivity.java
package com.photo.editor.picskills.photoeditorpro.activities;


import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.view.Window;
import android.widget.Button;
import android.widget.ImageView;
import com.photo.editor.picskills.photoeditorpro.R;
import jp.shts.android.storiesprogressview.StoriesProgressView;

public class StatusActivity extends AppCompatActivity implements StoriesProgressView.StoriesListener, View.OnTouchListener, View.OnClickListener {
    private ImageView imageView, arrowTop;
    private StoriesProgressView storiesProgressView;
    private long pressTime = 0L;
    long limit = 500L;
    private Button tryNow;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        setContentView(R.layout.activity_status);
        imageView = findViewById(R.id.status_image);
        arrowTop = findViewById(R.id.arrow_top);
        tryNow = findViewById(R.id.try_now);
        tryNow.setOnClickListener(this);
        arrowTop.setOnClickListener(this);
        storiesProgressView = findViewById(R.id.stories);
        setModelStory();
        storiesProgressView.setStoriesCount(1); // <- set stories
        storiesProgressView.setStoryDuration(10000); // <- set a story duration  1200L
        storiesProgressView.setStoriesListener(this); // <- set listener
        storiesProgressView.startStories(); // <- start progress
        imageView.setOnTouchListener(this);
        arrowTop.setOnClickListener(this);
    }

    private void setModelStory() {
        if (MainActivity.isModelSelected == 0) {
            imageView.setImageDrawable(ContextCompat.getDrawable(
                    this,
                    R.drawable.wings_status
            ));
        }
        if (MainActivity.isModelSelected == 1) {
            imageView.setImageDrawable(ContextCompat.getDrawable(
                    this,
                    R.drawable.spiral_status
            ));
        }
        if (MainActivity.isModelSelected == 2) {
            imageView.setImageDrawable(ContextCompat.getDrawable(
                    this,
                    R.drawable.blur_status
            ));
        }
        if (MainActivity.isModelSelected == 3) {
            imageView.setImageDrawable(ContextCompat.getDrawable(
                    this,
                    R.drawable.b_and_w_status
            ));
        }
        if (MainActivity.isModelSelected == 4) {
            imageView.setImageDrawable(ContextCompat.getDrawable(
                    this,
                    R.drawable.filter_status
            ));
        }
        if (MainActivity.isModelSelected == 5) {
            imageView.setImageDrawable(ContextCompat.getDrawable(
                    this,
                    R.drawable.gradient_status
            ));
        }
    }

    @Override
    public void onNext() {
        Log.i("onNext", "next");
    }

    @Override
    public void onPrev() {
        Log.i("onPrev", "previous");
    }

    @Override
    public void onComplete() {
        finish();
    }

    @Override
    protected void onDestroy() {
        // Very important !
        storiesProgressView.destroy();
        super.onDestroy();
    }

    @Override
    public boolean onTouch(View v, MotionEvent event) {
        if (v.getId() == R.id.status_image) {
            switch (event.getAction()) {
                case MotionEvent.ACTION_DOWN:
                    pressTime = System.currentTimeMillis();
                    storiesProgressView.pause();
                    return true;
                case MotionEvent.ACTION_UP:
                    long now = System.currentTimeMillis();
                    storiesProgressView.resume();
                    return limit < now - pressTime;
            }
        }
        return false;
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.try_now) {
            itemClick();
        }
        if (v.getId() == R.id.arrow_top) {
            itemClick();
        }
    }

    private void itemClick() {
        if (MainActivity.isModelSelected == 0) {
            finish();
            MainActivity.getActivity().showIdOrPermission();
        }
        if (MainActivity.isModelSelected == 1) {
            finish();
            MainActivity.getActivity().showIdOrPermission();
        }
        if (MainActivity.isModelSelected == 2) {
            finish();
            MainActivity.getActivity().showIdOrPermission();
        }
        if (MainActivity.isModelSelected == 3) {
            finish();
            MainActivity.getActivity().showIdOrPermission();
        }
        if (MainActivity.isModelSelected == 4) {
            finish();
            MainActivity.getActivity().showIdOrPermission();
        }
        if (MainActivity.isModelSelected == 5) {
            finish();
            MainActivity.getActivity().showIdOrPermission();
        }
    }
}

File: app/src/main/java/com/photo/editor/picskills/photoeditorpro/activities/StatusActivity.kt
package com.photo.editor.picskills.photoeditorpro.activities

import com.photo.editor.picskills.photoeditorpro.activities.MainActivity.Companion.activity
import androidx.appcompat.app.AppCompatActivity
import jp.shts.android.storiesprogressview.StoriesProgressView.StoriesListener
import android.view.View.OnTouchListener
import android.os.Bundle
import android.util.DisplayMetrics
import android.util.Log
import com.photo.editor.picskills.photoeditorpro.R
import androidx.core.content.ContextCompat
import android.view.MotionEvent
import android.view.View
import android.view.Window
import com.google.android.gms.ads.AdRequest
import com.google.android.gms.ads.AdSize
import com.google.android.gms.ads.AdView
import com.photo.editor.picskills.photoeditorpro.databinding.ActivityStatusBinding
import com.photo.editor.picskills.photoeditorpro.utils.UtilsPak
import java.lang.Exception

class StatusActivity : AppCompatActivity(), StoriesListener, OnTouchListener, View.OnClickListener {

    private var pressTime = 0L
    var limit = 500L
    lateinit var mainBinding: ActivityStatusBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        mainBinding = ActivityStatusBinding.inflate(layoutInflater)
        setContentView(mainBinding.root)

        bannerAds()

        mainBinding.tryNow.setOnClickListener(this)
        mainBinding.arrowTop.setOnClickListener(this)
        setModelStory()
        mainBinding.stories.setStoriesCount(1) // <- set stories
        mainBinding.stories.setStoryDuration(10000) // <- set a story duration  1200L
        mainBinding.stories.setStoriesListener(this) // <- set listener
        mainBinding.stories.startStories() // <- start progress
        mainBinding.statusImage.setOnTouchListener(this)
        mainBinding.arrowTop.setOnClickListener(this)
    }

    private fun setModelStory() {
        if (MainActivity.isModelSelected == 0) {
            mainBinding.statusImage.setImageDrawable(
                ContextCompat.getDrawable(
                    this,
                    R.drawable.wings_status
                )
            )
        }
        if (MainActivity.isModelSelected == 1) {
            mainBinding.statusImage.setImageDrawable(
                ContextCompat.getDrawable(
                    this,
                    R.drawable.spiral_status
                )
            )
        }
        if (MainActivity.isModelSelected == 2) {
            mainBinding.statusImage.setImageDrawable(
                ContextCompat.getDrawable(
                    this,
                    R.drawable.blur_status
                )
            )
        }
        if (MainActivity.isModelSelected == 3) {
            mainBinding.statusImage.setImageDrawable(
                ContextCompat.getDrawable(
                    this,
                    R.drawable.b_and_w_status
                )
            )
        }
        if (MainActivity.isModelSelected == 4) {
            mainBinding.statusImage.setImageDrawable(
                ContextCompat.getDrawable(
                    this,
                    R.drawable.filter_status
                )
            )
        }
        if (MainActivity.isModelSelected == 5) {
            mainBinding.statusImage.setImageDrawable(
                ContextCompat.getDrawable(
                    this,
                    R.drawable.gradient_status
                )
            )
        }
    }

    override fun onNext() {
        Log.i("onNext", "next")
    }

    override fun onPrev() {
        Log.i("onPrev", "previous")
    }

    override fun onComplete() {
        finish()
    }

    override fun onDestroy() {
        // Very important !
        mainBinding.stories.destroy()
        super.onDestroy()
    }

    override fun onTouch(v: View, event: MotionEvent): Boolean {
        if (v.id == R.id.status_image) {
            when (event.action) {
                MotionEvent.ACTION_DOWN -> {
                    pressTime = System.currentTimeMillis()
                    mainBinding.stories.pause()
                    return true
                }
                MotionEvent.ACTION_UP -> {
                    val now = System.currentTimeMillis()
                    mainBinding.stories.resume()
                    return limit < now - pressTime
                }
            }
        }
        return false
    }

    override fun onClick(v: View) {
        if (v.id == R.id.try_now) {
            itemClick()
        }
        if (v.id == R.id.arrow_top) {
            itemClick()
        }
    }

    private fun itemClick() {
        if (MainActivity.isModelSelected == 0) {
            finish()
            activity!!.showIdOrPermission()
        }
        if (MainActivity.isModelSelected == 1) {
            finish()
            activity!!.showIdOrPermission()
        }
        if (MainActivity.isModelSelected == 2) {
            finish()
            activity!!.showIdOrPermission()
        }
        if (MainActivity.isModelSelected == 3) {
            finish()
            activity!!.showIdOrPermission()
        }
        if (MainActivity.isModelSelected == 4) {
            finish()
            activity!!.showIdOrPermission()
        }
        if (MainActivity.isModelSelected == 5) {
            finish()
            activity!!.showIdOrPermission()
        }
    }

    /**************************************Banner Ads *********************************/
    fun bannerAds() {
        mainBinding.adLayout.visibility = View.VISIBLE
        mainBinding.adLayout.post { loadBanner() }
    }

    private fun loadBanner() {
        Log.e("myTag", "BannerAds")
        val adView = AdView(this)
        adView.adUnitId = UtilsPak.getBannerID()
        val adSize = adSize
        adView.setAdSize(adSize)
        mainBinding.adLayout.addView(adView)
        val adRequest = AdRequest.Builder().build()
        // Start loading the ad in the background.
        try {
            adView.loadAd(adRequest)
        } catch (ex: Exception) {
            ex.printStackTrace()
        }
    }

    // Determine the screen width (less decorations) to use for the ad width.
    @Suppress("DEPRECATION")
    private val adSize: AdSize
        get() {
            // Determine the screen width (less decorations) to use for the ad width.
            val display = windowManager.defaultDisplay
            val outMetrics = DisplayMetrics()
            display.getMetrics(outMetrics)
            val density = outMetrics.density
            var adWidthPixels = mainBinding.adLayout.width.toFloat()
            // If the ad hasn't been laid out, default to the full screen width.
            if (adWidthPixels == 0f) {
                adWidthPixels = outMetrics.widthPixels.toFloat()
            }
            val adWidth = (adWidthPixels / density).toInt()
            return AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(this, adWidth)
        }
}